### PR TITLE
Cleanup/issue 207 scrape and replace namespace level "using" statements

### PR DESCRIPTION
**Pull request by sakrejda**
Locally passes all math unit tests, I assume this means I didn't leave a meaningless name anywhere...  this only catches the stray namespace-level 'using' statements that follow within five lines of a namespace statement so it might miss some.

### CODE REVIEW

File: stan/math/prim/mat/err/check_cholesky_factor_corr.hpp [hunk 1]
```diff
@@ -37,9 +36,12 @@ namespace stan {
...
     template <typename T_y>
     bool
-    check_cholesky_factor_corr(const char* function,
-                               const char* name,
-                               const Eigen::Matrix<T_y, Dynamic, Dynamic>& y) {
+    check_cholesky_factor_corr(
+      const char* function,
+      const char* name,
+      const Eigen::Matrix<T_y, Eigen::Dynamic, Eigen::Dynamic>& y
+    ) {
```
Comment (bob-carpenter) [inline stan/math/prim/mat/err/check_cholesky_factor_corr.hpp]: This closing paren and open bracket should be on the previous line.

File: stan/math/prim/mat/err/check_corr_matrix.hpp [hunk 2]
```diff
@@ -43,13 +42,16 @@ namespace stan {
...
     template <typename T_y>
     inline bool
-    check_corr_matrix(const char* function,
-                      const char* name,
-                      const Eigen::Matrix<T_y, Dynamic, Dynamic>& y) {
+    check_corr_matrix(
+      const char* function,
+      const char* name,
+      const Eigen::Matrix<T_y, Eigen::Dynamic, Eigen::Dynamic>& y
+    ) {
```
Comment (bob-carpenter) [inline stan/math/prim/mat/err/check_corr_matrix.hpp]: Same --- it was right before in terms of layout.

File: stan/math/prim/mat/err/check_cov_matrix.hpp [hunk 3]
```diff
@@ -29,9 +28,11 @@ namespace stan {
...
     template <typename T_y>
     inline bool
-    check_cov_matrix(const char* function,
-                     const char* name,
-                     const Eigen::Matrix<T_y, Dynamic, Dynamic>& y) {
+    check_cov_matrix(
+      const char* function,
+      const char* name,
+      const Eigen::Matrix<T_y, Eigen::Dynamic, Eigen::Dynamic>& y
+    ) {
```
Comment (bob-carpenter) [inline stan/math/prim/mat/err/check_cov_matrix.hpp]: Same.

File: stan/math/prim/mat/err/check_lower_triangular.hpp [hunk 4]
```diff
@@ -32,9 +31,11 @@ namespace stan {
...
     template <typename T_y>
     inline bool
-    check_lower_triangular(const char* function,
-                           const char* name,
-                           const Eigen::Matrix<T_y, Dynamic, Dynamic>& y) {
+    check_lower_triangular(
+      const char* function,
+      const char* name,
+      const Eigen::Matrix<T_y, Eigen::Dynamic, Eigen::Dynamic>& y
+    ) {
```
Comment (bob-carpenter) [inline stan/math/prim/mat/err/check_lower_triangular.hpp]: If it doesn't fit, you can break under the T_y, and start the Eigen::Dynamic lined up under T_y on the next line.

File: stan/math/prim/mat/err/check_pos_definite.hpp [hunk 5]
```diff
@@ -35,9 +34,11 @@ namespace stan {
...
     template <typename T_y>
     inline bool
-    check_pos_definite(const char* function,
-                       const char* name,
-                       const Eigen::Matrix<T_y, Dynamic, Dynamic>& y) {
+    check_pos_definite(
+      const char* function,
+      const char* name,
+      const Eigen::Matrix<T_y, Eigen::Dynamic, Eigen::Dynamic>& y
+    ) {
```
Comment (bob-carpenter) [inline stan/math/prim/mat/err/check_pos_definite.hpp]: Same.

File: stan/math/prim/mat/err/check_pos_semidefinite.hpp [hunk 6]
```diff
@@ -33,9 +32,11 @@ namespace stan {
...
     template <typename T_y>
     inline bool
-    check_pos_semidefinite(const char* function,
-                           const char* name,
-                           const Eigen::Matrix<T_y, Dynamic, Dynamic>& y) {
+    check_pos_semidefinite(
+      const char* function,
+      const char* name,
+      const Eigen::Matrix<T_y, Eigen::Dynamic, Eigen::Dynamic>& y
+    ) {
```
Comment (bob-carpenter) [inline stan/math/prim/mat/err/check_pos_semidefinite.hpp]: One more.

File: stan/math/prim/mat/err/check_spsd_matrix.hpp [hunk 7]
```diff
@@ -29,9 +28,11 @@ namespace stan {
...
     template <typename T_y>
     inline bool
-    check_spsd_matrix(const char* function,
-                      const char* name,
-                      const Eigen::Matrix<T_y, Dynamic, Dynamic>& y) {
+    check_spsd_matrix(
+      const char* function,
+      const char* name,
+      const Eigen::Matrix<T_y, Eigen::Dynamic, Eigen::Dynamic>& y
+    ) {
```
Comment (bob-carpenter) [inline stan/math/prim/mat/err/check_spsd_matrix.hpp]: I guess there's a lot of them.  I'll let you find the rest if you don't mind changing them.  If you don't want to change them, it's no big deal.  Maybe other people don't agree, but I find the line that starts with a close paren odd and hard to parse.

**Comment by bob-carpenter** [comment]
Thanks so much for finding all these and fixing them.  It all looks great.  I have some stylistic comments, but no big deal if you don't want to follow them.

**Comment by sakrejda** [comment]
No problem, it's a fine way to get familiar with different corners of the library.  I was splitting the lines like that to keep cpplint happy... if I just move the paren/brace back up like you're suggesting I end up with the function arguments horizontally aligned with the function body and no indicator when I'm going from one to the other.  I guess I don't know how to make it readable any other way---your suggestion works sometimes but if you have long and meaningful function names you're pushing the arguments arbitrarily far right... so for example, would you prefer something like:

~~~
  template <typename F>
    void
    grad_tr_mat_times_hessian(const F& f,
                              const Eigen::Matrix<double, Eigen::Dynamic, 1>& x,
                              const Eigen::Matrix<double, Eigen::Dynamic, 
                                Eigen::Dynamic>& M,
                              Eigen::Matrix<double, 
                                Eigen::Dynamic, 1>& grad_tr_MH) {
      using Eigen::Matrix;
      using Eigen::Dynamic;
...
~~~

over this:

~~~
  template <typename F>
    void
    grad_tr_mat_times_hessian( 
      const F& f,
      const Eigen::Matrix<double, Eigen::Dynamic, 1>& x,
      const Eigen::Matrix<double, Eigen::Dynamic, Eigen::Dynamic>& M,
      Eigen::Matrix<double, Eigen::Dynamic, 1>& grad_tr_MH
    ) {

~~~

I could try to make your style work when I deal with running cpplint locally (it seems to not run through on the server when it tallies up enough complaints to fail) but at this point I just don't get what you're going for.

**Comment by bob-carpenter** [comment]
> On Dec 2, 2015, at 12:39 AM, Krzysztof Sakrejda <notifications@github.com> wrote:
> 
> No problem, it's a fine way to get familiar with different corners of the library. I was splitting the lines like that to keep cpplint happy...

Understood.  It requires 80 characters max.

> if I just move the paren/brace back up like you're suggesting I end up with the function arguments horizontally aligned with the function body and no indicator when I'm going from one to the other. I guess I don't know how to make it readable any other way---your suggestion works sometimes but if you have long and meaningful function names you're pushing the arguments arbitrarily far right... so for example, would you prefer something like:
> 

I like this:

>     template <typename F>
>     void grad_tr_mat_times_hessian(const F& f,
>                       const Eigen::Matrix<double, Eigen::Dynamic, 1>& x,
>                       const Eigen::Matrix<double, Eigen::Dynamic, 
>                       Eigen::Dynamic>& M,
>                       Eigen::Matrix<double, Eigen::Dynamic, 1>& grad_tr_MH) {
> ...
> 
> over this:
> 
>     template <typename F>
>     void
>     grad_tr_mat_times_hessian( 
>       const F& f,
>       const Eigen::Matrix<double, Eigen::Dynamic, 1>& x,
>       const Eigen::Matrix<double, Eigen::Dynamic, Eigen::Dynamic>& M,
>       Eigen::Matrix<double, Eigen::Dynamic, 1>& grad_tr_MH
>     ) {

> I could try to make your style work when I deal with running cpplint locally (it seems to not run through on the server when it tallies up enough complaints to fail) but at this point I just don't get what you're going for.

It's pretty easy to express:

  * don't start lines with ")"  --- this is the biggy

  * don't end lines with "(" --- this one I don't mind breaking

It follows from the "pack as much on a line as possible" principle,
which in turn follows from the "vertical space is precious" principle.
My version is 6 lines, yours is 8.  You read where you're at by
indentation.

It's not even just my style --- read the Google C++ style guide for both
examples and explanation as to why they like that style.  It's very
much in line with what Java went with in its own libraries, which is where
I picked it up ages ago.

If you don't feel like changing it all, it's no big deal --- as long
as cpplint passes.  The problem we have is that the existing code isn't
consistently styled, which makes it impossible to figure out what to do
by example.

- Bob

**Comment by sakrejda** [comment]
Ok, I'll have to spend some more time with the style guide to be able to do this consistently.  I'd like to skip it for this PR.

**Comment by syclik** [comment]
Yes, that's fine. Thank you! We really appreciate the help. 

Movement in the right direction is always good.
